File: Code/Android/JsonAnalyze/app/src/main/java/me/ztiany/jsonanalyze/AppContext.java
package me.ztiany.jsonanalyze;

import android.app.Application;

/**
 * @author Ztiany
 * Date : 2018-08-13 11:14
 */
public class AppContext extends Application {

    private static AppContext sAppContext;

    @Override
    public void onCreate() {
        super.onCreate();
        sAppContext = this;
    }

    public static AppContext getAppContext() {
        return sAppContext;
    }

}

File: Android/00-Code/AAC-Sample/Paging-Sample/src/main/java/com/ztiany/acc/AppContext.kt
package com.ztiany.acc

import android.app.Application
import android.content.Context
import android.support.multidex.MultiDex

/**
 *
 *@author Ztiany
 *      Email: ztiany3@gmail.com
 *      Date : 2018-04-28 16:10
 */
class AppContext : Application() {

    companion object {
        private lateinit var INSTANCE: AppContext
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

    override fun onCreate() {
        super.onCreate()
        INSTANCE = this
    }

    fun getAppContext() = INSTANCE


}